testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -1107,6 +1092,17 @@
         )
 end
 
+# In case of a search reindex not having finished yet, keep retrying until successful search or timeout
+When(/^I click on the search button$/) do
+  click_button_and_wait('Search', match: :first)
+
+  repeat_until_timeout(message: 'Could not perform a successful search after reindexation', timeout: 10) do
+    break unless has_text?('Could not connect to search server.', wait: 0) || has_text?('No matches found', wait: 0)
+    sleep 1
```

Comment:
Why we wait 1 second?

Reply:
to avoid active wait ; all other calls return immediately

Reply:
In such case, we should move that wait after clicking the button, in that position it doesn't make sense.

Reply:
You need to test the result immediately after pressing the button.